### PR TITLE
[EDX-129] New design for site top nav

### PR DESCRIPTION
**Pull request by MymmiJ**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

This is to get the docs site top nav looking like this Figma representation: https://www.figma.com/file/vyr9ZPZ15eUWtCKSEktdEL/API-refs?node-id=414%3A8643

* [Jira ticket](https://ably.atlassian.net/browse/EDX-129).
* [Search bar design](https://ably.atlassian.net/browse/EDX-137).

## Review

Instructions on how to review the PR. 

* [Page to review](http://localhost:8000/docs/api/realtime-sdk) - any page should work, check the top nav and ensure it looks as expected.
* If you can run the website with `bin/dev` (it only really depends on postgresql and redis + rake jobs), you can test logging in and viewing your account details in that dropdown menu.

### CODE REVIEW

**Review by DuncanEdwards — CHANGES_REQUESTED** [review]
Before I even start on the code there are probably layout issues to resolve:
image

Also, is the LHS column going to be a fixed width? In the design the search is right next to the logo, but in the implementation that LHS column can grow, taking the Search box away from the logo. Is that OK? Would Lenka be happy with that?

**Comment by MymmiJ** [comment]
I wasn't sure what to do when the page spaces out more, I think it looks better the way you propose - I can cluster the search + docs to the left and everything else to the right.

**Comment by MymmiJ** [comment]
> Before I even start on the code there are probably layout issues to resolve: ...
> 
> Also, is the LHS column going to be a fixed width? In the design the search is right next to the logo, but in the implementation that LHS column can grow, taking the Search box away from the logo. Is that OK? Would Lenka be happy with that?

I've made these changes, the logo & search button/dropdown menu items now cluster together.

**Review by DuncanEdwards — COMMENTED** [review]
Seem to be missing top and bottom delimiter lines:
image

Whereas on the website meganav it looks different:
image

**Comment by MymmiJ** [comment]
> Seem to be missing top and bottom delimiter lines: image
> 
> Whereas on the website meganav it looks different: image

I've had a go at changing these to look more like the website meganav.

File: src/components/Header/TopMainNav/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { DropdownDataIdentifier } from './Dropdown/types';
+import { TopMainNavUserMenu } from './TopMainNavUser';
+import UserContext from '../../../contexts/user-context';
+import { HorizontalMenuItemGroup } from '../../Menu/HorizontalMenuItemGroup';
+
+export const TopMainNav = () => {
+  const [dropdownDataID, setDropdownDataID] = useState<DropdownDataIdentifier>(null);
+  const setDropdownData = (value: DropdownDataIdentifier) => () => setDropdownDataID(value);
+  const clearDropdownData = () => setDropdownDataID(null);
+  const menuItems: (keyof typeof dropdownData)[] = ['API References', 'Resources'];
```
Comment (DuncanEdwards) [inline src/components/Header/TopMainNav/index.tsx:17]: index!!

**Review by DuncanEdwards — CHANGES_REQUESTED** [review]
Minor things + break out `index.ts`

**Review by DuncanEdwards — APPROVED** [review]

**Pull request merged**